corehq/apps/smsbillables/utils.py

```diff
@@ -92,3 +93,34 @@ def _get_infobip_message_details(api_channel, api_suffix, config, headers, param
     response_content = json.loads(response.content)
     messages = response_content['results']
     return messages
+
+
+@quickcache(vary_on=['backend_instance', 'backend_message_id'], timeout=1 * 60)
+def get_pinpoint_message(backend_instance, backend_message_id):
+    from corehq.messaging.smsbackends.amazon_pinpoint.models import PinpointBackend
+    try:
+        pinpoint_backend = SQLMobileBackend.load(
+            backend_instance,
+            api_id=PinpointBackend.get_api_id(),
+            is_couch_id=True,
+            include_deleted=True,
+        )
+        pinpoint_service = 'logs'
+        config = pinpoint_backend.config
+        client = _get_pinpoint_client(backend_instance, pinpoint_service)
+        response = client.filter_log_events(
+            logGroupName=f'sns/{config.region}/754026553166/DirectPublishToPhoneNumber',
```

Comment:
What is this magic number "754026553166"? Seems like that belongs in a configuration option somewhere.

The [AWS Pinpoint gateway package](https://github.com/dimagi/commcare-hq/tree/1a2abc4070bc095c8139d25a828bcacd939f8583/corehq/messaging/smsbackends/amazon_pinpoint) seems like a better place for this code than this utils module.